spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/support/DefaultPublisherFactory.java

```diff
@@ -53,11 +55,17 @@
 
 	private final ChannelProvider channelProvider;
 
+	private final CredentialsProvider credentialsProvider;
+
 	public DefaultPublisherFactory(String projectId, ExecutorProvider executorProvider,
-			ChannelProvider channelProvider) {
+			CredentialsProvider credentialsProvider) {
 		this.projectId = projectId;
 		this.executorProvider = executorProvider;
-		this.channelProvider = channelProvider;
+		this.channelProvider = TopicAdminSettings.defaultChannelProviderBuilder()
```

Comment:
I understand that the `credentialsProvider` no longer has to be set on the `channelProvider`, but why would we not want to allow the users to configure their own `channelProvider` and keep its auto-configuration?

In other words, why not have the constructor take both `ChannelProvider` and `CredentialsProvider`?